### PR TITLE
Fix: Display issue with the free to pro promotion in the structure panel [ED-14153]

### PR DESCRIPTION
**Pull request by Batsirai-muchareva**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: assets/dev/scss/editor/navigator/navigator.scss [hunk 1]
```diff
@@ -113,9 +110,10 @@ body {
 		&__promotion {
 			font-size: 12px;
 			padding: 4px 17px;
-			line-height: 22px;
 			font-family: 'Roboto', sans-serif;
 			color: var(--e-a-bg-logo);
+			min-height: 30px;
```
Comment (TzviRabinovitch) [inline assets/dev/scss/editor/navigator/navigator.scss:115]: Maybe use vars ?
@louiswol94 wdyt ?

File: assets/dev/scss/editor/navigator/navigator.scss [hunk 2]
```diff
@@ -113,9 +110,10 @@ body {
...
+			line-height: 18px;
```
Comment (hein-obox) [inline assets/dev/scss/editor/navigator/navigator.scss:116]: Why has the line height changed?

File: assets/dev/scss/editor/navigator/navigator.scss [hunk 3]
```diff
@@ -91,12 +93,7 @@ body {
...
 	}
 
-	&__elements.elementor-navigator-list__promotion {
-		height: calc(100% - 80px);
-	}
-
 	&__footer {
-		position: absolute;
 		width: 100%;
 		left: 0;
```
Comment (hein-obox) [inline assets/dev/scss/editor/navigator/navigator.scss:98]: Can we remove some of these properties, now that we have removed `position: absolute`?

File: includes/editor-templates/panel-elements.php [hunk 4]
```diff
@@ -42,7 +42,9 @@
 	<?php if ( ! $has_pro ) : ?>
 	<div id="elementor-panel-get-pro-elements-sticky">
 		<img class="elementor-nerd-box-icon" src="<?php echo ELEMENTOR_ASSETS_URL . 'images/unlock-sticky.svg'; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>" loading="lazy" alt="<?php echo esc_attr__( 'Upgrade', 'elementor' ); ?>"/>
-		<div class="elementor-get-pro-sticky-message"><?php echo esc_html( $promotion_data_sticky['message'] ); ?><a target="_blank" href="<?php echo esc_url( $promotion_data_sticky['url'] ); ?>"><?php echo esc_html( $promotion_data_sticky['button_text'] ); ?></a></div>
+		<div class="elementor-get-pro-sticky-message">
```
Comment (hein-obox) [inline includes/editor-templates/panel-elements.php:45]: Did we have to change this structure? Or has this been done for readability?

File: tests/playwright/sanity/modules/promotions/promotions.test.ts [hunk 5]
```diff
@@ -155,4 +155,22 @@ test.describe( 'Promotion tests @promotions', () => {
...
+		// Act.
+		for ( let i = 0; i < 20; i++ ) {
+			await editor.addElement( { elType: 'container' }, 'document' );
+		}
+
+		await navigatorPanel.locator( '.elementor-navigator__element-container' ).nth( 0 ).click();
+		await navigatorPanel.evaluate( ( element ) => element.style.width = '150px' );
+
+		// Assert.
+		await expect( navigatorPanel ).toHaveScreenshot( 'resized-navigator-panel.png' );
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/promotions/promotions.test.ts:174]: Nice.

File: assets/dev/scss/editor/navigator/navigator.scss (hunk 2 above)
Comment (hein-obox) [reply assets/dev/scss/editor/navigator/navigator.scss:116]: We try to keep the number of changes in a PR as low as possible, to make the review easier.

Comment (louiswol94) [reply assets/dev/scss/editor/navigator/navigator.scss:116]: @Batsirai-muchareva Can we rather use `display: flex;` with `align-items: center;`? Not sure if line-height was the best approach to begin with for controlling the alignment.

File: assets/dev/scss/editor/navigator/navigator.scss (hunk 1 above)
Comment (louiswol94) [reply assets/dev/scss/editor/navigator/navigator.scss:115]: Yeah, not sure if `min-height` is needed.

File: includes/editor-templates/panel-elements.php (hunk 4 above)
Comment (louiswol94) [reply includes/editor-templates/panel-elements.php:45]: If this was done for readability, the closing `</div>` should also be on the next line.

File: assets/dev/scss/editor/navigator/navigator.scss (hunk 2 above)
Comment (louiswol94) [reply assets/dev/scss/editor/navigator/navigator.scss:116]: I know Hadar probably approved it, but to me, the new screenshots look even less vertically aligned than before 🤔 WDYT?

File: assets/dev/scss/editor/navigator/navigator.scss [hunk 6]
```diff
@@ -1,5 +1,6 @@
 body {
 	--e-editor-navigator-width: 240px;
+	--e-editor-navigator-promotion-height: 30px;
```
Comment (hein-obox) [inline assets/dev/scss/editor/navigator/navigator.scss:3]: Screenshots are failing.

File: assets/dev/scss/editor/navigator/navigator.scss [hunk 7]
```diff
@@ -113,12 +109,16 @@ body {
...
 			font-family: 'Roboto', sans-serif;
 			color: var(--e-a-bg-logo);
+			min-height: var(--e-editor-navigator-promotion-height);
+			display: flex;
+			justify-content: center;
+			align-items: center;
 
 			a {
 				color: var(--e-a-color-accent-promotion);
+				margin-inline-start: 1px;
```
Comment (hein-obox) [inline assets/dev/scss/editor/navigator/navigator.scss:121]: What is this for?

Comment (Batsirai-muchareva) [reply assets/dev/scss/editor/navigator/navigator.scss:121]: text and link spacing

Comment (hein-obox) [reply assets/dev/scss/editor/navigator/navigator.scss:121]: Huddle?

**Review by hein-obox — APPROVED** [review]

**Pull request merged**